l3kernel/l3file.dtx

```diff
@@ -1499,10 +1499,21 @@
 %
 % \begin{macro}{\@@_new:N}
 %   As for read streams, copy \tn{newwrite}, making sure
-%   that it is not \tn{outer}.
+%   that it is not \tn{outer}. For \ConTeXt{}, we have to
+%   deal with the fact that \tn{newwrite} works like our
+%   own: it actually checks before altering definition.
 %    \begin{macrocode}
 \exp_args:NNf \cs_new_protected:Npn \@@_new:N
   { \exp_args:NNc \exp_after:wN \exp_stop_f: { newwrite } }
+\cs_if_exist:NT \normalend
+  {
+    \cs_new_eq:NN \@@_new_aux:N \@@_new:N
+    \cs_set_protected:Npn \@@_new:N #1
```

Comment:
Should be `gset`

Reply:
in format mode it doesn't really matter I guess, but if somebody uses this as a blueprint to detect ConTeXt elsewhere, then there is the problem that `\normalend` is simply an ordinary name and could get defined as a user command --- not likely but with just ascii letters not impossible either.

Reply:
Fixed in [bcd1d1f](https://github.com/latex3/latex3/pull/1114/commits/bcd1d1faa28c3768f3cbc372e9baf9f94a6a91dd).